deps/chakrashim/src/jsrtinspectorhelpers.cc

```diff
@@ -578,35 +638,197 @@ namespace jsrt {
+      CHAKRA_VERIFY_NOERROR(jsrt::InspectorHelpers::GetProperty(stackProperties, "functionCallsReturn", &functionCallsReturn));
+      InspectorHelpers::ArrayConcat(localsArray, functionCallsReturn);
+    }
+
+    CHAKRA_VERIFY_NOERROR(jsrt::InspectorHelpers::HasProperty(stackProperties, "locals", &hasProp));
+
+    if (hasProp) {
+      JsValueRef locals;
+      CHAKRA_VERIFY_NOERROR(jsrt::InspectorHelpers::GetProperty(stackProperties, "locals", &locals));
+      InspectorHelpers::ArrayConcat(localsArray, locals);
+    }
+
+    return jsrt::InspectorHelpers::WrapPropertiesArray(localsArray);
+  }
+
+  v8::Local<v8::Value> InspectorHelpers::EvaluateOnCallFrame(
+      int ordinal, JsValueRef expression, bool returnByValue, bool* isError) {
+    CHAKRA_VERIFY(ordinal >= 0);
+
+    if (isError != nullptr) {
+      *isError = false;
+    }
+
+    JsValueRef evalResult;
+    JsErrorCode err = JsDiagEvaluate(expression, ordinal,
+                                     JsParseScriptAttributeNone,
+                                     returnByValue, &evalResult);
+
+    if (err == JsErrorScriptException) {
+      if (isError != nullptr) {
+        *isError = true;
+      }
+
+      return WrapException(evalResult);
+    }
+    else if (err == JsErrorDiagNotAtBreak) {
+      return v8::Local<v8::Value>();
+    }
+
+    CHAKRA_VERIFY_NOERROR(err);
+    return WrapEvaluateObject(evalResult);
+  }
+
+  v8::Local<v8::Value> InspectorHelpers::EvaluateOnCallFrame(
+      JsValueRef callFrame, JsValueRef expression, bool returnByValue,
+      bool* isError) {
+    CHAKRA_VERIFY(callFrame != nullptr);
+
+    int ordinal;
+    CHAKRA_VERIFY_NOERROR(InspectorHelpers::GetIntProperty(callFrame, "index", &ordinal));
+
+    return EvaluateOnCallFrame(ordinal, expression, returnByValue, isError);
   }
 
   JsRuntimeHandle InspectorHelpers::GetRuntimeFromIsolate(
       v8::Isolate *isolate) {
     return IsolateShim::FromIsolate(isolate)->GetRuntimeHandle();
   }
 
+  void InspectorHelpers::WrapObject(JsValueRef obj, JsValueRef* wrappedObj) {
+    JsValueRef value;
+    CHAKRA_VERIFY_NOERROR(JsCreateObject(&value));
+
+    CHAKRA_VERIFY(CopyPropertyIfPresent(obj, "type", value));
+
+    CopyPropertyIfPresent(obj, "className", value);
+    
+    bool hasValue = CopyPropertyIfPresent(obj, "value", value);
+    bool hasDisplay = CopyPropertyIfPresent(obj, "display", value, "description");
+
+    // A description is required for values to be shown in the debugger.
+    if (hasValue && !hasDisplay) {
+      hasDisplay = CopyPropertyStringIfPresent(obj, "value", value, "description");
+    }
+
+    CHAKRA_VERIFY(hasDisplay);
+
+    int handle;
+    CHAKRA_VERIFY_NOERROR(InspectorHelpers::GetIntProperty(obj, "handle", &handle));
+    CHAKRA_VERIFY_NOERROR(InspectorHelpers::SetStringProperty(value, "objectId", GetObjectIdForHandle(handle).c_str()));
+
+    *wrappedObj = value;
+  }
+
+  void InspectorHelpers::WrapProperty(JsValueRef propValue, JsValueRef* wrappedProperty) {
```

Comment:
>WrapProperty [](start = 25, length = 12)

again, see if this can be done in shim.js. The reason I am pushing for this is multiple Jsrt calls has penalty at the boundary. See [here](https://github.com/nodejs/node-chakracore/blob/efeaf88f0c734fb3837608654b356e9eb0f9ae64/deps/chakrashim/core/lib/Jsrt/JsrtInternal.h#L153). If we shim in `.js` file, it will be 1 Jsrt API call.